### PR TITLE
Script to import log files into s3 bucket

### PR DESCRIPTION
**Pull request by yashita-accolite**
The script saves the required log files to s3 bucket,
 (For now, it does not contain script to save xxx_build.txt and xxx_compile.txt)

### CODE REVIEW

File: Scripts/import-logs.ps1 [hunk 1]
```diff
@@ -0,0 +1,110 @@
+#########powershell script to upload log files into s3 bucket###########
+
+#s3Bucket path to save the log files
+param (
+    [string] $version,
+    [string] $ReleaseId
+)
+$Lansaversion = $version
```
Comment (robe070) [inline Scripts/import-logs.ps1]: I can see no value in having local copies of the parameters. Please rename the parameters to be the Local names and remove these 2 lines.
As in
```
param (
    [string] $Lansaversion,
    [string] $PipelineReleaseId 
)
```

You also need to receive the List of Database Types that are being tested and only pick up THOSE log files.

File: Scripts/import-logs.ps1 [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+$PipelineReleaseId = $ReleaseId
+$s3BasePath = 'lansa-us-east-1/db-regression-test/logs' 
+$region = 'us-east-1'
+ 
+#Total Summary for all Dbs
+$totalSummaryFile = 'C:\Program Files (x86)\LANSA\Verifier_Total_Summary.txt'
+$totalSummaryPath = Join-Path $Lansaversion $PipelineReleaseId
+$totalSummaryKey = $totalSummaryPath + '/Verifier_Total_Summary.txt'
+Write-S3Object -BucketName $s3BasePath -Key $totalSummaryKey -Content $totalSummaryFile
```
Comment (robe070) [inline Scripts/import-logs.ps1]: You need to use log messages. This is not possible to diagnose issues right now.
Use Write-Host as described in Teams.
So for example, here you need to log `$totalSummaryKey ` and `$totalSummaryFile`.
Also add `| Out-Default | Write-Host` to the end of the `Write-S3Object` line.

With log messages you could simply provide those to me to answer the questions in the Issue about SPECIFIC paths.

File: Scripts/import-logs.ps1 [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+#Total Summary for all Dbs
+$totalSummaryFile = 'C:\Program Files (x86)\LANSA\Verifier_Total_Summary.txt'
+$totalSummaryPath = Join-Path $Lansaversion $PipelineReleaseId
+$totalSummaryKey = $totalSummaryPath + '/Verifier_Total_Summary.txt'
+Write-S3Object -BucketName $s3BasePath -Key $totalSummaryKey -Content $totalSummaryFile 
+
+
+################################ list of databases ################################################################
+$databases = New-Object -TypeName 'System.Collections.ArrayList';
+ #$databases.Add("SQL Server")
```
Comment (robe070) [inline Scripts/import-logs.ps1]: I see your problem with SQL Server. That is NOT the path to it. Its `LANSA`.

File: Scripts/import-logs.ps1 [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+ $databases.Add("AZURESQL")
+ $databases.Add("Oracle")
+ $databases.Add("MySQL")
```
Comment (robe070) [inline Scripts/import-logs.ps1]: MySQL is not setup fully yet. So there are no log files to find.

File: Scripts/import-logs.ps1 [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+ 
+#Total Summary for all Dbs
+$totalSummaryFile = 'C:\Program Files (x86)\LANSA\Verifier_Total_Summary.txt'
+$totalSummaryPath = Join-Path $Lansaversion $PipelineReleaseId
+$totalSummaryKey = $totalSummaryPath + '/Verifier_Total_Summary.txt'
+Write-S3Object -BucketName $s3BasePath -Key $totalSummaryKey -Content $totalSummaryFile 
+
+
+################################ list of databases ################################################################
+$databases = New-Object -TypeName 'System.Collections.ArrayList';
```
Comment (robe070) [inline Scripts/import-logs.ps1]: The style here is good. And you need to be adding in databases based on the Database Types passed in.

**Review by robe070 — CHANGES_REQUESTED** [review]
So not a single log message in here. This is very important. The person diagnosing this will not be on the VM. Very removed from it. You must tell them what you are attempting and the result of what you attempted.

You also need a Try/Catch block.

File: Scripts/import-logs.ps1 (hunk 5 above)
Comment (yashita-accolite) [reply Scripts/import-logs.ps1]: Hi @robe070 I have seen your comments, will address those and get back

**Comment by robe070** [comment]
Also note that `DB2ISERIES` will not have any log files of its own. Its a special case. Its log files will be in the SQL Server system - `Root = C:\Program Files (x86)\lansa`

**Comment by robe070** [comment]
So the LANSA Systems to expect log files in at the moment are:
LANSA
SQLANYWHERE
AZURESQL

**Comment by yashita-accolite** [comment]
Hi @robe070 , I have made the suitable changes, please review.

File: Scripts/import-logs.ps1 [hunk 6]
```diff
@@ -1,109 +1,259 @@
...
+Write-Host('list of databases:') 
+Write-Host('--------------------------------------------------------------')
+foreach ($db in $databaseTypes)
+{
+    if($db -ne 'SQL Server')
+    {
+        $databases.Add($db)
+    }  
+    Write-Host($db) 
+}
```
Comment (robe070) [inline Scripts/import-logs.ps1:39]: The database types are here: https://github.com/lansa/db-regression/wiki#:~:text=Limited%20to%20these,n/a
`SQL Server` is not one of them. Its `MSSQLS`.
You need to map from DB Type to RootPath.

File: Scripts/import-logs.ps1 [hunk 7]
```diff
@@ -1,109 +1,259 @@
...
-$compiledObjectspath = 'x_win95\x_lansa\X_WBP\Compile\DBTEST\CompiledObjects.txt'
-foreach ($database in $databases)
-{
-    
-    $databaseRootPath = Join-Path $RootPath $database
-	$compiledObjectsfile = Join-Path $databaseRootPath $compiledObjectspath
-    $key = $totalSummaryPath + '/' + $database + '/CompiledObjects.txt'
-    Write-S3Object -BucketName $s3BasePath -Key $key -File $compiledObjectsfile -Region $region
+try{
+    Write-Host('COMPILEDOBJECTS.TXT')
```
Comment (robe070) [inline Scripts/import-logs.ps1]: Looks OK. Except, you have 3 loops for 3 separate files.
Change it to 1 loop which copies ALL the files in DBTEST folder to S3.

File: Scripts/import-logs.ps1 [hunk 8]
```diff
@@ -1,109 +1,259 @@
...
+    Write-Host('#Verifier_Test_Summary.txt')
+    Write-Host('--------------------------------------------------------------')
+    $verifierTestSummary = 'Verifier_Test_Summary.txt'
+    foreach ($database in $databases)
+    {
+        $databaseRootPath = Join-Path $RootPath $database
+        $verifierTestSummaryFile = Join-Path $databaseRootPath $verifierTestSummary
+        $key=''
+        if($database -eq 'Lansa'){
+            $key = $s3BaseKey  + '/MYSQLS/Verifier_Test_Summary.txt'
```
Comment (robe070) [inline Scripts/import-logs.ps1]: You use this MYSQLS folder in various places. Why? See the DB Types. I think it should be MSSQLS.

**Review by robe070 — CHANGES_REQUESTED** [review]
Looking at S3:db-regression-test/logs/111/1234/SQLANYWHERE/&showversions=false
You need folders on here to reflect where they came from. E.g. CompiledObjects.txt in a DBTEST folder.
*_Compile.txt files in CompileDetails folder.
The verifier*.txt files can remain at the top level as they are a summary.

File: Scripts/import-logs.ps1 (hunk 7 above)
Comment (yashita-accolite) [reply Scripts/import-logs.ps1]: changed the logic to save all files from dbtest, (DBTEST also has skippedObjects.txt, which is getting saved now)

File: Scripts/import-logs.ps1 (hunk 8 above)
Comment (yashita-accolite) [reply Scripts/import-logs.ps1]: done

File: Scripts/import-logs.ps1 [hunk 9]
```diff
@@ -17,107 +17,61 @@ $region = 'us-east-1'
...
 {
-    if($db -ne 'SQL Server')
-    {
-        $databases.Add($db)
-    }  
+    
+    if($db -ne 'DB2ISERIES'){
+        $databases.Add($DB_ROOT_Map[$db])
+    }     
     Write-Host($db)
```
Comment (robe070) [inline Scripts/import-logs.ps1:38]: Log both $db and $DB_ROOT_Map[$db]. The latter is far more important. Its what the code is deriving.

**Review by robe070 — CHANGES_REQUESTED** [review]

**Pull request merged**